lib/open_project/plugins/acts_as_op_engine.rb

```diff
@@ -68,6 +68,18 @@ def self.included(base)
         end
       end
 
+      base.send(:define_method, :patch_with_namespace) do |*args|
+        plugin_name = engine_name
+        base.config.to_prepare do
+          klass_name = args.last
+          plugin_module = plugin_name.sub(/^openproject_/, '').camelcase
+          patch = "OpenProject::#{plugin_module}::Patches::#{klass_name}Patch".constantize
+          qualified_class_name = args.map(&:to_s).join('::')
+          klass = qualified_class_name.to_s.constantize
+          klass.send(:include, patch) unless klass.included_modules.include?(patch)
+        end
+      end
```

Comment:
You can get the patches constant easier and more robust like this:

```
plugin_module = self.class.to_s.deconstantize
base.config.to_prepare do
  klass_name = args.last
  patch = "#{plugin_module}::Patches::#{klass_name}Patch".constantize
  ....
end
```

And while you are at it: **you should not copy code**